### PR TITLE
fix: update amd loader for better debugging and understandability

### PR DESCRIPTION
**Pull request by menduz**
# What?

Proper names and error catching in amd-loader

# Why?

There was several scenes halting not knowing why. This PR fixes some bugs I fixed a while ago in [ulla](https://github.com/arduz-io/ulla) and adds a lot of readability to the code.

+ Also eliminates obscure names and make things easier to read.
+ Also compiles with strict mode

### CODE REVIEW

File: kernel/packages/decentraland-amd/tsconfig.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "es2017",
```
Comment (eordano) [inline kernel/packages/decentraland-amd/tsconfig.json:3]:
```suggestion
    "target": "es5",
```

File: kernel/packages/decentraland-amd/tsconfig.json [hunk 2]
```diff
@@ -13,9 +13,18 @@
     "experimentalDecorators": true,
     "emitDecoratorMetadata": true,
     "noUnusedLocals": true,
-    "lib": ["es5"],
+    "strict": true,
+    "strictNullChecks": true,
+    "lib": [
+      "es2017"
```
Comment (eordano) [inline kernel/packages/decentraland-amd/tsconfig.json:19]:
```suggestion
      "es5"
```

File: kernel/packages/decentraland-amd/package.json [hunk 3]
```diff
@@ -12,11 +12,17 @@
   "files": [
     "dist"
   ],
+  "prettier": {
+    "semi": true,
```
Comment (eordano) [inline kernel/packages/decentraland-amd/package.json]:
```suggestion
    "semi": false,
```

File: kernel/packages/decentraland-amd/package.json [hunk 4]
```diff
@@ -12,11 +12,17 @@
...
+    "singleQuote": false,
```
Comment (eordano) [inline kernel/packages/decentraland-amd/package.json]:
```suggestion
    "singleQuote": true,
```

File: kernel/packages/decentraland-amd/tsconfig.json (hunk 1 above)
Comment (menduz) [reply kernel/packages/decentraland-amd/tsconfig.json:3]: It uses sets and array.from, there is no real reason to stick to ES5 anymore

File: kernel/packages/decentraland-amd/package.json (hunk 3 above)
Comment (menduz) [reply kernel/packages/decentraland-amd/package.json]: I did not want to change the styles that were already in place for the file. may that be another pr?

File: kernel/packages/decentraland-amd/package.json (hunk 4 above)
Comment (menduz) [reply kernel/packages/decentraland-amd/package.json]: same

File: kernel/packages/decentraland-amd/package.json (hunk 3 above)
Comment (eordano) [reply kernel/packages/decentraland-amd/package.json]: then let's rather drop this section

**Comment by menduz** [comment]
Styles fixed. What do we resolve with the ES6 features?

@nearnshaw had his scene fixed but raised a concern about other scene getting broken because of this. Could you Nico post more details here?

**Comment by nearnshaw** [comment]
I tried this on a few existing scenes, and it solved problems on some but raised new ones on others.

Specifically, I tried this out on the Genesis Plaza and Soho Plaza scenes, that both have a lot of different files importing from each other, but that currently work well.
In both cases the scene stoped loading using this. The console claimed that it couldn't load any of the modules, saying that perhaps there were circular dependencies. 
In the code there are dependencies between different .ts files, and there were also libraries that were referenced  several times from multiple files, but no circular dependencies between objects, systems, etc.

https://github.com/decentraland-scenes/Genesis-Plaza
https://github.com/decentraland-scenes/Soho-Plaza

image


We can't merge this if it breaks existing scenes like that, there may be many affected scenes that don't get fixed.

File: kernel/packages/decentraland-amd/tsconfig.json (hunk 1 above)
Comment (eordano) [reply kernel/packages/decentraland-amd/tsconfig.json:3]: Thanks! That's related to the `lib`, right? Let's leave the `target` as es5

**Pull request closed without merging**